rlp/typecache.go

```diff
@@ -101,7 +117,10 @@ func structFields(typ reflect.Type) (fields []field, err error) {
 }
 
 func parseStructTag(tag string) tags {
-	return tags{nilOK: tag == "nil"}
+	return tags{
+		nilOK:  strings.Contains(tag, "nil"),
+		dotdot: strings.Contains(tag, ".."),
+	}
```

Comment:
I think it would catch potential errors if these would do strict checks and not `contains`. I.e. split the `tag` by commas, and check each element for `nil`, or `...` and also error out if there's something else in there.